File: Algorithm/BOJ/11723.cpp
//https://www.acmicpc.net/problem/11723
//http://1ilsang.blog.me/221446095017

#include <cstdio>
#include <string>

using namespace std;

int main(void) {
    int t;
    int bit = 0;
    string cmd;
    scanf("%d", &t);

    while (t--) {
        int tmp = 0, n;
        char str[50];
        scanf("%s", str);
        cmd = str;
        if (cmd == "all") bit = (1 << 21) - 1;
        else scanf(" %d", &n);

        if (cmd == "add") bit |= (1 << (n - 1));
        else if (cmd == "remove") bit &= ~(tmp | 1 << (n - 1));
        else if (cmd == "check") (bit & (1 << (n - 1))) ? puts("1") :puts("0");
        else if (cmd == "toggle") bit = bit ^ (1 << (n - 1));
        else if (cmd == "empty") bit = 1 >> 21;
    }
    return 0;
}

File: Algorithm/BOJ/java/11723.java
//https://www.acmicpc.net/problem/11723

import java.io.*;

public class Main {

    public static void main(String[] args) throws IOException {
        int list = 0;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            String[] tmp = br.readLine().split(" ");
            int num;
            if (tmp[0].equals("all")) list = (1 << 21) - 1;
            else if (tmp[0].equals("empty")) list = 1 >> 21;
            else {
                num = Integer.parseInt(tmp[1]);
                if (tmp[0].equals("add")) list |= 1 << num;
                else if (tmp[0].equals("remove")) list &= ~(1 << num);
                else if (tmp[0].equals("check")) bw.write(((list & (1 << num)) > 0 ? 1 : 0) + "\n");
                else list ^= 1 << num;
            }
        }
        bw.close();
    }
}
